### PR TITLE
Fix: Building death effects

### PR DESCRIPTION
**Pull request by Stubbjax**
Applied a bunch of fixes to various building scaffold death effects, and tweaked some death effect sizes. The reason for the missing effects was due to the custom 'explosion' death replacements for the respective buildings, which are exempt from playing while `UNDER_CONSTRUCTION`. The equivalent effects for the `UNDER_CONSTRUCTION` condition were evidently overlooked.

- Scud Storms now play an appropriate effect when the scaffold is cancelled.
- Nuclear Reactors now play an appropriate effect when the scaffold is cancelled.
- Nuclear Silos now play an appropriate effect when the scaffold is cancelled.
- Reduced the size of America Barracks death effects (medium → small).
- Reduced the size of Firebase death effects (medium → small).
- Reduced the size of Supply Drop Zone death effects (medium → small).

## 1.04:
When the Scud Storm scaffold is cancelled, there is no proper effect.

https://user-images.githubusercontent.com/11547761/196043627-d2a31e85-6cab-44ee-a114-bf0b0534d6f9.mp4

## Patch:
When the Scud Storm scaffold is cancelled, there is a proper effect.

https://user-images.githubusercontent.com/11547761/196043635-8ccd21d2-1a08-45bf-9b38-0c643fbee5af.mp4

### CODE REVIEW

**Comment by xezon** [comment]
Did you review all structures?

**Comment by commy2** [comment]
Missing Boss, GC Stealth and GC Tox Scud Storms. Also missing Boss Nuclear Missile Silo.

**Comment by commy2** [comment]
lgtm

Maybe someone make a table which building maps to which effect? E.g, the patch changes USA Barracks to small. Does that mean the other Barracks also use small?

**Review by commy2 — APPROVED** [review]

**Comment by xezon** [comment]
I tested. My notes:

* GLA SCUD Storm spawns no debris
* China Propaganda Center spawns no debris
* USA Firebase spawns too much debris, compared to GLA Tunnel, Stinger
* China Bunker spawns too much debris, compared to GLA Tunnel, Stinger
* China Propaganda Tower spawns too much debris, compared to GLA Tunnel, Stinger
* USA Patriot spawns no debris, just a tiny explosion puff, which probably is ok if there is no smaller debris we can use
* China Gattling Cannon spawns too much and too large debris. It is small structure, so should look like USA Patriot

**Comment by Stubbjax** [comment]
> I tested. My notes:

I have addressed all of this and more. There were (and likely still are) a lot more issues than I originally thought. I think the latest changes will have to suffice for now before the PR becomes too large.

> Maybe someone make a table which building maps to which effect? E.g, the patch changes USA Barracks to small. Does that mean the other Barracks also use small?

That's probably what I'll have to end up doing. But yes, I made sure there was more consistency between the factions - all Barracks now use small.

**Review by xezon — APPROVED** [review]
Looks good. Perhaps can be fine tuned more afterwards, but the foundation looks very solid now.

File: Patch104pZH/GameFilesEdited/Data/INI/Object/ChemicalGeneral.ini [hunk 1]
```diff
@@ -5693,6 +5693,10 @@ Object Chem_GLAScudStorm
     DeathFX       = FX_StructureMediumDeath
   End
 
+  Behavior  = CreateObjectDie ModuleTag_20
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/ChemicalGeneral.ini]:
```suggestion
  Behavior  = CreateObjectDie ModuleTag_22
```
To keep this number consistent between all variants.

File: Patch104pZH/GameFilesEdited/Data/INI/Object/DemoGeneral.ini [hunk 2]
```diff
@@ -5009,6 +5009,10 @@ Object Demo_GLAScudStorm
     DeathFX       = FX_StructureMediumDeath
   End
 
+  Behavior  = CreateObjectDie ModuleTag_20
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/DemoGeneral.ini]:
```suggestion
  Behavior  = CreateObjectDie ModuleTag_22
```

File: Patch104pZH/GameFilesEdited/Data/INI/Object/FactionBuilding.ini [hunk 3]
```diff
@@ -13559,6 +13559,10 @@ Object GLAScudStorm
     DeathFX       = FX_StructureMediumDeath
   End
 
+  Behavior  = CreateObjectDie ModuleTag_20
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/FactionBuilding.ini:13562]:
```suggestion
  Behavior  = CreateObjectDie ModuleTag_22
```

File: Patch104pZH/GameFilesEdited/Data/INI/Object/GC_Chem_GLABuildings.ini [hunk 4]
```diff
@@ -8399,6 +8399,10 @@ Object GC_Chem_GLAScudStorm
     DeathFX       = FX_StructureMediumDeath
   End
 
+  Behavior  = CreateObjectDie ModuleTag_20
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/GC_Chem_GLABuildings.ini]:
```suggestion
  Behavior  = CreateObjectDie ModuleTag_22
```

File: Patch104pZH/GameFilesEdited/Data/INI/Object/GC_Slth_GLABuildings.ini [hunk 5]
```diff
@@ -8673,6 +8673,11 @@ Object GC_Slth_GLAScudStorm
     RequiredStatus = UNDER_CONSTRUCTION
     DeathFX       = FX_StructureMediumDeath
   End
+
+  Behavior  = CreateObjectDie ModuleTag_21
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/GC_Slth_GLABuildings.ini]:
```suggestion
  Behavior  = CreateObjectDie ModuleTag_22
```

File: Patch104pZH/GameFilesEdited/Data/INI/Object/StealthGeneral.ini [hunk 6]
```diff
@@ -5743,6 +5743,10 @@ Object Slth_GLAScudStorm
     DeathFX       = FX_StructureMediumDeath
   End
 
+  Behavior  = CreateObjectDie ModuleTag_20
```
Comment (commy2) [inline Patch104pZH/GameFilesEdited/Data/INI/Object/StealthGeneral.ini]:
```suggestion
  Behavior  = CreateObjectDie ModuleTag_22
```

**Review by commy2 — APPROVED** [review]

**Review by xezon — APPROVED** [review]

**Pull request merged**